### ISSUE TITLE
[UX] Improve the default user management view and account edit form with links to manage permissions/roles

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5918]
## Use case

(This makes more sense on complex/bigger sites with many user accounts and multiple roles, but still applies to simpler sites)

A person that has an account on the site contacts you (being the site admin or the developer that is looking after their site), and requests that they get access to perform task x. What usually happens is that the site admin will:

1. go to the the user account management view at `/admin/people/list`
2. filter the list by the person's name/email etc.
3. examine the role(s) the person has, then determine if:
   - the person needs an existing role assigned to them in addition to the ones they already have (a role that already has the permission to do task x)
   - or if a new role needs to be created (because the existing roles that have that permission are overly permissive, with more permissions than what the person should have)
   - or an existing role needs to be edited to add the permission that was requested
4. depending on the case form the previous step, the site admin will need to do any of the following:
   - a. additional role assignment: click the existing "edit" button in the user account screen, add/remove existing roles as required -> done 👍🏼 
   - b. new role needs to be created: need to head to the roles management page at `/admin/config/people/roles` (see #3814, ??? and other issues relevant to improving that workflow)
   - c. permissions of an existing role need to be updated: need to head to `/admin/config/people/permissions` to edit the permissions of the existing user role(s) assigned to the user

## Proposed improvement

Since the other workflows in point 4 above are either being handled OK already, or we have existing issues for them, I would like us streamline the last workflow a bit (case 4c), so that there's less clicking around, less page loads, and less back and forth between admin pages. I propose that we consider implementing the following:
1. Make the role names in the "roles" column of the view in `/admin/people/list` links that take you directly to the permissions management form, filtered by the respective role column only. Alternatively, we could add a single "manage permissions" link to each user account, which filter the permissions management page only by the role(s) already assigned to the user.
2. Either also make the roles in the account edit form links (not a big fan of this, since these are labels of checkboxes), or add a set of two edit links right after the checkboxes: a "manage permissions" and an "edit roles".

**Comment by klonos** [comment]
...the above might be confusing to read through, so I will also provide before/after screenshots for each of the proposals. Bear with me.

**Comment by klonos** [comment]
...here's what I have basically on my local when an admin assigns roles to an account:
<img src="https://user-images.githubusercontent.com/2423362/211431724-fda374ae-2abd-47e1-9e2b-1c51d8f7ff22.png" width=65%>

And here's what non-admin users, with only the permission to manage accounts and assign roles will get (only descriptions - no links):
<img src="https://user-images.githubusercontent.com/2423362/211431789-d531350a-0656-4c2d-9de4-6f9b2889617a.png" width=45%>

**Comment by klonos** [comment]
...I've tested the idea of linkifying the role names in the user account management view, but I'm not really happy with it. What should the link do? ...configure the role, or manage the permissions? As it currently is, an admin will need to click the "edit" link for an account, and they'll go to the edit form, where they can find the links 🤷🏼 

Anyone has any better ideas re links in the account listing view?

**Comment by klonos** [comment]
...here's a PR with a sandbox for people to play with: https://github.com/backdrop/backdrop/pull/4302

**Comment by klonos** [comment]
@docwilmot suggested the following in the PR:

> Maybe open in dialog might be slicker than new tab/window?

Curious to see what others think about this. I'm not opposed to it, but not sure how dialogs would work on mobile (although, I very frequently wonder who manages their site on mobile). Also, the role management form would be fine in a dialog, not so much the huge permissions matrix 🤔

**Comment by olafgrabienski** [comment]
@klonos You wrote this feature request makes more sense on complex or bigger sites with many user accounts and multiple roles. I'd say it makes only sense on sites where roles and/or permissions are changed frequently. In my experience, this is quite uncommon (or usually only the case in the initial development phase of a site). So, as the links to roles and permissions configuration are only needed in rare cases, I don't see them as an improvement. In my opinion, they rather clutter the UI.

**Comment by kiamlaluno** [comment]
IMO, the role description is more helpful when the roles are reconsidered, when a new role needs to be added, or when a role needs to be assigned to users. In those cases, knowing what the purpose of the existing roles is helps.  
Clearly, usually the role names are self-descriptive, but that cannot "catch" all the nuances.

Some sites could have roles that are kept for old users, but that should not be used for new users. In this case, a description that says the role should not be given to new users would be much helpful.

Personally, I have never needed to role descriptions, but in the case there are more users who can create roles, that could be helpful. This is also the case of a site created by somebody and then maintained by somebody else; in that case, documenting the different roles directly in the site is helpful.

**Comment by klonos** [comment]
Fair enough @olafgrabienski and @kiamlaluno ...perhaps this is better left for contrib or something. I would personally like to see things like #4643 addressed.

I'll leave this issue and PR open a while longer, to give others the opportunity to chime in, then will most likely close them.

**Comment by kiamlaluno** [comment]
I think this feature should be implemented in Backdrop core. I won't install a module just to be able to add role descriptions.

I like what shown in the last screenshots. What I would eventually leave to another issue is adding the *Configure role* and *Manage permissions* links.

**Comment by klonos** [comment]
> I think this feature should be implemented in Backdrop core. ...

@kiamlaluno yes, the descriptions have already been added into Backdrop core with #5748 and will be available in the upcoming 1.24.0 release after January 15. The reason why you see that in my screenshots is because I am using the most recent 1.x clone of the codebase on my local, which includes this new feature.

> What I would eventually leave to another issue or a contributed module is the *Configure role* and *Manage permissions* links.

Yes again. This issue here is specifically for adding the *Configure role* and *Manage permissions* links. This is what I am saying now (after the last comments from Olaf and you) that we should leave for contrib.

**Comment by kiamlaluno** [comment]
Those links are fine for me, but if other users think those links clutter the page, I would find a way to hide those links and show them in the same way *Hide row weights* and *Show row weights* do with row weight fields.

**Comment by klonos** [comment]
I think that it's worth pointing out that these links will only be shown for people that have the `administer permissions` permission - so content editors/approvers and people that have simply been tasked to manage user accounts on the site (people with the `administer users` and `assign roles` permissions) won't see these links at all.

**Comment by jenlampton** [comment]
I like the idea of adding help text to the top of the user account list page. It could contain a link to the user permissions page, which can be hard to find. But I don't think we need to add links from that page to the user account list since that's a top-level link in the admin menu. 

I don't particularly like adding these links to Roles checkboxes area. It will clutter the form and adjusting permissions isn't usually something that's changed at the same time as a user is assigned a role (unless there is a larger problem with site configuration).

**Comment by klonos** [comment]
>I like the idea of adding help text to the top of the user account list page. ...

Yes 👍🏼 ...I got the feeling that we all agreed that that would be best when we discussed this during the UX/design meeting.

> ...It could contain a link to the user permissions page, which can be hard to find.

And another link to the roles management page. Right?
